Better support parallel Spicy installations

Right now, when Spicy is installed through Homebrew, it's tricky to teach a source build of the plugin to *not* use that Homebrew version, but pick up Spicy in some other location. I'm seeing trouble mixing up includes and libraries, even when the right spicy-config is being found. Haven't fully tracked it down yet what's going on.

We should tweak the CMake config to make it straight-forward to point to a specific Spicy version to use. Best would be indeed to derive everything from `spicy-config` (which is happening already) and then prioritize that over paths found differently (which doesn't seem to happen yet).

At least for the linking step, I can reproduce this as follows. First, I have Spicy 1.1 installed through Homebrew. Then I manually compile a Spicy 1.3, install that into a local prefix, and put that `bin/` into my `PATH`. The plugin's cmake then finds 1.3 as expected, but still attempts to link against 1.1. Transcript below.

```
# which zeek
/Users/robin/work/zeek/install/4.0/bin/zeek

# which spicy-config
/usr/local/bin/spicy-config

# spicy-config --version
1.1.0 (72d88fbc)

# export PATH=$HOME/work/spicy/demo/bin:$PATH
# which spicy-config
/Users/robin/work/spicy/demo/bin/spicy-config

# spicy-config --version
1.3.0 (3872229d)

# mkdir build && cd build && cmake ..
-- The CXX compiler identification is AppleClang 13.0.0.13000029
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found spicy-config: /Users/robin/work/spicy/demo/bin/spicy-config
-- Found Spicy: yes
-- Found zeek-config: /Users/robin/work/spicy/demo/bin/zeek-config
-- Found Zeek: yes
-- Performing Test cxx17_already_works
-- Performing Test cxx17_already_works - Success
-- Found BinPAC: /Users/robin/work/spicy/demo/bin/binpac
-- Found CAF: 0.18.2  found components: core io openssl
-- Found Broker: /Users/robin/work/spicy/demo/lib/libbroker.dylib
-- Zeek executable      : /Users/robin/work/spicy/demo/bin/zeek
-- Zeek source          :
-- Zeek build           :
-- Zeek install prefix  : /Users/robin/work/spicy/demo
-- Zeek plugin directory: /Users/robin/work/spicy/demo/lib/zeek/plugins
-- Zeek debug mode      :

====================|  Spicy Installation Summary  |====================

Found Spicy:           yes

Version:               1.3.0 (3872229d) (10300)
Prefix:                /Users/robin/work/spicy/demo
Build type:            release
Have toolchain:        yes
Spicy compiler:        /Users/robin/work/spicy/demo/bin/spicyc

========================================================================


====================|  Spicy-side Zeek Installation Summary  |====================

Found Zeek:            yes

Version:               4.0.4 (40004)
Prefix:                /Users/robin/work/spicy/demo
Build type:            relwithdebinfo

========================================================================


====================|  Spicy Zeek Plugin  |====================

Version:               1.3.1 (10301)
Build type:            RelWithDebInfo
Modules directory:     /usr/local/lib/zeek-spicy/modules
Scripts directory:     /usr/local/lib/zeek-spicy/scripts
Build directory:       /Users/robin/work/spicy/demo/src/spicy-plugin/build
Have JIT:              yes
Zeek debug build:      no
Zeek-internal build:   no
spicy-config:          /Users/robin/work/spicy/demo/bin/spicy-config
zeek-config:           /Users/robin/work/spicy/demo/bin/zeek-config

========================================================================

-- Configuring done
-- Generating done
-- Build files have been written to: /Users/robin/work/spicy/demo/src/spicy-plugin/build

# make VERBOSE=1
[...]
[ 96%] Linking CXX executable ../../bin/spicyz
cd /Users/robin/work/spicy/demo/src/spicy-plugin/build/src/compiler && /usr/local/Cellar/cmake/3.20.5/bin/cmake -E cmake_link_script CMakeFiles/spicyz.dir/link.txt --verbose=1
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -I /usr/local/include -O3 -g -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk -mmacosx-version-min=11.6 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/usr/local/lib CMakeFiles/spicyz.dir/bin/spicyz.cc.o CMakeFiles/zeek-compiler.dir/driver.cc.o CMakeFiles/zeek-compiler.dir/glue-compiler.cc.o -o ../../bin/spicyz   -L/Users/robin/work/spicy/demo/lib  -Wl,-rpath,/Users/robin/work/spicy/demo/lib -lhilti -lspicy
Undefined symbols for architecture x86_64:
  "hilti::Unit::fromModule(std::__1::shared_ptr<hilti::Context> const&, hilti::Module, ghc::filesystem::path)", referenced from:
      spicy::zeek::GlueCompiler::compile() in glue-compiler.cc.o
  "hilti::node::none", referenced from:
      std::__1::vector<hilti::Node, std::__1::allocator<hilti::Node> > hilti::nodes<std::__1::optional<hilti::statement::detail::Statement> >(std::__1::optional<hilti::statement::detail::Statement>) in glue-compiler.cc.o
      hilti::to_node(std::__1::optional<hilti::AttributeSet>&&) in glue-compiler.cc.o
      hilti::declaration::GlobalVariable::GlobalVariable(hilti::ID, hilti::expression::detail::Expression, hilti::declaration::Linkage, hilti::Meta) in glue-compiler.cc.o
      hilti::statement::If::If(hilti::expression::detail::Expression, hilti::statement::detail::Statement, std::__1::optional<hilti::statement::detail::Statement>, hilti::Meta) in glue-compiler.cc.o
      hilti::statement::Return::Return(hilti::Meta) in glue-compiler.cc.o
      hilti::type::tuple::Element::Element(hilti::Type, hilti::Meta) in glue-compiler.cc.o
      std::__1::vector<hilti::Node, std::__1::allocator<hilti::Node> > hilti::nodes<std::__1::optional<hilti::expression::detail::Expression> >(std::__1::optional<hilti::expression::detail::Expression>) in glue-compiler.cc.o
      ...
  "hilti::type::detail::isResolved(hilti::Type const&, std::__1::set<unsigned long, std::__1::less<unsigned long>, std::__1::allocator<unsigned long> >*)", referenced from:
      hilti::type::Function::_isResolved(std::__1::set<unsigned long, std::__1::less<unsigned long>, std::__1::allocator<unsigned long> >*) const in glue-compiler.cc.o
      hilti::ctor::Tuple::_inferType(std::__1::vector<hilti::expression::detail::Expression, std::__1::allocator<hilti::expression::detail::Expression> > const&) in glue-compiler.cc.o
      hilti::type::detail::Model<hilti::type::Tuple>::_isResolved(std::__1::set<unsigned long, std::__1::less<unsigned long>, std::__1::allocator<unsigned long> >*) const in glue-compiler.cc.o
      hilti::type::detail::Model<hilti::type::vector::Iterator>::_isResolved(std::__1::set<unsigned long, std::__1::less<unsigned long>, std::__1::allocator<unsigned long> >*) const in glue-compiler.cc.o
      hilti::type::detail::Model<hilti::type::Vector>::_isResolved(std::__1::set<unsigned long, std::__1::less<unsigned long>, std::__1::allocator<unsigned long> >*) const in glue-compiler.cc.o
  "hilti::Driver::addInput(std::__1::shared_ptr<hilti::Unit>)", referenced from:
      spicy::zeek::GlueCompiler::compile() in glue-compiler.cc.o
  "hilti::Driver::Driver(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)", referenced from:
      spicy::zeek::Driver::Driver(char const*, ghc::filesystem::path, int) in driver.cc.o
  "hilti::declaration::ImportedModule::properties() const", referenced from:
      hilti::declaration::detail::Model<hilti::declaration::ImportedModule>::properties() const in glue-compiler.cc.o
  "hilti::type::Enum::uniqueLabels() const", referenced from:
      hilti::type::Enum::typeParameters() const in driver.cc.o
  "hilti::type::Enum::labels() const", referenced from:
      spicy::zeek::GlueCompiler::compile() in glue-compiler.cc.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Note the `-L/usr/local/lib` coming before `-L/Users/robin/work/spicy/demo/lib`.

--- a/zkg.meta
+++ b/zkg.meta
@@ -2,7 +2,7 @@
 script_dir = scripts/Zeek/Spicy
 plugin_dir = build
 
-build_command = mkdir -p build && cd build && cmake .. && make -j "${SPICY_ZKG_PROCESSES:-4}"
-test_command = cd tests && btest -d -j "${SPICY_ZKG_PROCESSES:-4}"
+build_command = unset -v CXX CXXFLAGS LD LDFLAGS && mkdir -p build && cd build && cmake .. && make -j "${SPICY_ZKG_PROCESSES:-4}"
+test_command = unset -v CXX CXXFLAGS LD LDFLAGS && cd tests && btest -d -j "${SPICY_ZKG_PROCESSES:-4}"
 
 executables = build/bin/spicyz

